### ISSUE TITLE
"_restart" console command crashes server

### ISSUE THREAD

**Issue by doctorluk** [Facepunch/garrysmod-issues#2406]
After manually updating my server (no *validate*) with the January update, it has since been crashing every day.

Upon further exploration, I've found out that my daily cronjob to restart the server has since been crashing it instead of restarting it.

This is the console's output upon sending the *_restart* command:

*_restart
SoundEmitter:  removing map sound overrides [5 to remove, 0 to restore]
L 01/22/2016 - 18:16:22: server_message: "quit"
L 01/22/2016 - 18:16:22: Log file closed.
L 01/22/2016 - 18:16:22: server_message: "restart"
Fri Jan 22 18:16:22 CET 2016: Server Quit*

My server is running TTT and has been running and restarting fine for over a year now, but the recent update broke this function.

**Comment by robotboy655** [comment]
It doesn't crash the server as far as I can tell, it just shuts it down completely.

**Comment by doctorluk** [comment]
Yeah, wrong wording from my side. It shuts the server down instead of restarting it.

**Comment by Bo98** [comment]
#1593 claims to have a crash dump

**Comment by robotboy655** [comment]
That is more than a year old, there WAS a crash when quitting the server with workshop addons, it was fixed.

**Comment by doctorluk** [comment]
My server is running on Ubuntu 14.04.3 LTS if that helps.

I'm using shared folders for multiple instances of GMod to reduce disk usage, but I don't expect any of that to cause such a bug.

**Comment by MStruntze** [comment]
This has been the behavior for as long as I recall. At least on linux. I've been using this as well to perform daily restarts, but ever since I started using it, it has shut down the server just like Robotboy says. Are you sure you've only had this behavior after the most recent update? I don't mind the way it works, but the name is indeed a bit misleading.

**Comment by doctorluk** [comment]
This has been working flawlessly for a while now. My cronjob looks as follows:

`0 6 * * * /some/path/cronjob_restart_gmod.sh`

which sends this:

`screen -S screenname -p 0 -X stuff "_restart$(printf \\r)"`

**Comment by willox** [comment]
Maybe your setup restarts the server when it crashes and now that _restart doesn't crash, it doesn't restart?

I wasn't aware that _restart ever worked.

**Comment by doctorluk** [comment]
My setup doesn't do any magic behind the scenes. My gamepanel starts the server process within a screen, but doesn't bring restart capabilities. So I have been doing this by sending a *_restart* command. Sadly the panel loses track of the server's process if the screen is killed by something else than the panel (yeah, it's not very advanced... but it's free...)

**Comment by neico** [comment]
As far as I remember _restart was just like "exit" "quit" etc. in source1, but the fact that any of those commands crashed the server certainly eased work with automated scripts (even the default one that valve provided did the trick).

Changing it's behavior would probably only confuse people, especially when they come from other source1 games.

But having a command that actually does a reboot without shutting down the entire process would be a neat thing, though I would say that such a thing isn't as easy implemented as it sounds if you consider how the engine looks like internal...

**Comment by MStruntze** [comment]
A workaround to the current behaviour could be to run another screen with a script as such:
```shell
while true; do
if ! screen -list | grep -q "gmodsrvscreen"; then
   ./startgmodsrv.sh
fi
sleep 5;
done;
```

**Comment by UltimateByte** [comment]
@robotboy655 It's not a year since the quit command made the server crash :p 
https://github.com/Facepunch/garrysmod-issues/issues/2342

Or has it been fixed, then broken again since ? :p 

@doctorluk Not using LGSM yet ?

https://github.com/dgibbs64/linuxgsm
http://gameservermanagers.com/lgsm/gmodserver/

**Comment by robotboy655** [comment]
@UltimateByte Thanks for not bringing any useful information whatsoever. Please do not comment if you do not have anything to say.

The bug report https://github.com/Facepunch/garrysmod-issues/issues/1593 is over a year old. I never said the quit crash was fixed over a year ago. I fixed the crash in December, and it wasn't "broken again since".

**Comment by UltimateByte** [comment]
Sorry for misunderstanding that. I'll try to focus on diagnosing only in the future, i know you got loads of work already.

**Comment by EstevanTH** [comment]
I *exit* and *_restart* console commands
